service/customer/internal/db/database.go

```diff
@@ -0,0 +1,43 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/service/customer/internal/repo"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+)
+
+type database struct {
+	db *gorm.DB
+}
+
+func NewDatabase(url, schemaName string) (*database, error) {
+	db, err := gorm.Open(postgres.Open(url), &gorm.Config{
+		NamingStrategy: schema.NamingStrategy{
+			TablePrefix: fmt.Sprintf("%s.", schemaName),
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection to postgres db: %w", err)
+	}
+
+	if err = db.AutoMigrate(&repo.Customer{}); err != nil {
+		return nil, fmt.Errorf("failed to automigrate model 'customer': %w", err)
+	}
+
+	return &database{db: db}, nil
+}
+
+func (d *database) Create(value interface{}) error {
+	return d.db.Create(value).Error
+}
+
+func (d *database) Delete(value interface{}) error {
+	return d.db.Delete(value).Error
+}
+
+func (d *database) Where(query any, args ...any) repo.DB {
+	return &database{db: d.db.Where(query, args...)}
+}
```

Comment:
Seems like very general database)

Reply:
You mean it's better to get rid of this database class and allow repository to know about gorm library?

Reply:
Yes, it's ok that repo knows about gorm.
Your services know only about interface, so the realisation could be anything

Reply:
The problem of using interface is that only way to get error after some operation is to check value of field `Error` of *gorm.DB. So I don't know how to use interface is this case without such general database realisation.

Reply:
I've removed extra abstraction for DB due to your recommendation